crates/ruma-common/src/identifiers/matrix_uri.rs

```diff
@@ -298,26 +298,34 @@ impl MatrixToUri {
 
     /// Try parsing a `&str` into a `MatrixToUri`.
     pub fn parse(s: &str) -> Result<Self, Error> {
-        let without_base = if let Some(stripped) = s.strip_prefix(MATRIX_TO_BASE_URL) {
-            stripped
-        } else {
-            return Err(MatrixToError::WrongBaseUrl.into());
-        };
-
-        let url = Url::parse(MATRIX_TO_BASE_URL.trim_end_matches("#/"))
-            .expect("matrix.to base URL is valid")
-            .join(without_base)
-            .map_err(|_| MatrixToError::InvalidUrl)?;
-
-        let id = MatrixId::parse_with_sigil(url.path())?;
-        let mut via = vec![];
-
-        for (key, value) in url.query_pairs() {
-            if key.as_ref() == "via" {
-                via.push(value.parse()?);
-            } else {
-                return Err(MatrixToError::UnknownArgument.into());
-            }
+        let s = s.strip_prefix(MATRIX_TO_BASE_URL).ok_or(MatrixToError::WrongBaseUrl)?;
+        let s = s.strip_suffix('/').unwrap_or(s);
+
+        // Separate the identifiers and the query.
+        let mut parts = s.split('?');
+
+        let ids_part = parts.next().ok_or(MatrixIdError::NoIdentifier)?;
```

Comment:
FWIW this will never hit the error state, as a `Split` iterator will always yield at least one value.

Reply:
I know, I just didn't want to put an `unwrap()` or `expect()`. Maybe I should?

Reply:
imo an `.expect("a split iterator yields at least one value")` is okay here, if you're stating the invariant you're expecting.